### PR TITLE
fix: refactor usermenu and introduce business styles

### LINKED ISSUE #479: Bug: usermenu multiple issues
- allow controls to enable logged in state instead of activating by existence of user name -> userName and displayName is now possible to set logged in state
 - mobile usage (improved, but also see #647)
 - UserMenu contains a div which is overly wide. When flexing it on the right, it might overflow, causing a weird looking, horizontal scroll bar, forcing to set overflow-x to hidden.
- animation not ok. see sbb.ch (Franco) -> done
- refactoring code -> done
- better documentation that dropdown is neccesary to use (reported by Ghilardelli Marco) -> done (new docu)
- missing business styles -> done

```html
<sbb-usermenu [userName]="value" [displayName]="value">
  <button sbb-usermenu-item>Something 1</button>
  <a sbb-usermenu-item routerLink="/somewhere">Link 1</a>
  <hr>
  <button sbb-usermenu-item>Logout</button>
</sbb-usermenu>
```

### PR DESCRIPTION
**Pull request by jeripeierSBB**
This PR refactors the usermenu module in a technical and optical way, while also the API has slightly changed. 

BREAKING CHANGE: The usermenu module has been refactored. The `<sbb-usermenu>` now
considers both `displayName` and `userName` to set logged in state (you have to set at minimum 
one of both). 
The `userName` is not shown anymore in collapsed state of menu (according to digital.sbb.ch). Use 
`displayName` to show a name in collapsed state.
The usermenu doesn't depend on `<sbb-dropdown>` anymore, just use `<a sbb-usermenu-item>`, 
`<button sbb-usermenu-item>` and `<hr />` tags.
Providing a custom icon or image now needs to set the structural `*sbbIcon` directive instead of 
using only `sbbIcon`.

Closes #479

### CODE REVIEW

**Review by kyubisation — CHANGES_REQUESTED** [review]
I requested some changes in the migration.
Other than that this is great work! 👍

File: src/angular-public/usermenu/usermenu-item.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  host: {
+    role: 'menuitem',
+    class: 'sbb-usermenu-item',
+  },
+})
+export class SbbUsermenuItem implements FocusableOption {
+  constructor(private _focusMonitor: FocusMonitor, private _elementRef: ElementRef<HTMLElement>) {}
+
+  focus(origin?: FocusOrigin, options?: FocusOptions): void {
+    if (this._focusMonitor && origin) {
```
[the remark this answers is not in the document]
Comment (jeripeierSBB) [reply src/angular-public/usermenu/usermenu-item.ts:15]: the only reason is, that I copied it from sbb header. But there in the constructor the focusMonitor is optional. So yes, I remove it :)

File: src/angular-public/usermenu/usermenu.md [hunk 2]
```diff
@@ -1,92 +1,74 @@
-The usermenu is intended to log on / off and offers the logged-in user a menu
-with comprehensive functionalities.
+The usermenu is intended to log in / out a user and offers the logged-in user a menu
+with comprehensive functionalities. The Usermenu should be placed in a site header, e.g. the `<sbb-header>`.
```
Comment (kyubisation) [inline src/angular-public/usermenu/usermenu.md:2]:
```suggestion
with comprehensive functionalities. The usermenu should be placed in a site header, e.g. the `<sbb-header>`.
```

**Review by kyubisation — APPROVED** [review]
LGTM 👍

**Pull request merged**